test.py: only run `*.gen.sh` scripts actually generated during that run

kkysen opened the pull request:
Otherwise, a previously generated `*.gen.sh` file will still be run even if it's no longer supposed to be generated (and isn't anymore).

kkysen committed:
test.py: only run `*.gen.sh` scripts actually generated during that run

Otherwise, a previously generated `*.gen.sh` file will still be run
even if it's no longer supposed to be generated (and isn't anymore).

fw-immunant commented on tests/templates.py:
```diff
@@ -94,16 +95,16 @@ def render_script(template: str, out_path: str, params: Dict):
-def autogen_cargo(conf_file, yaml: Dict):
-    def render_stage(stage_conf: Mapping[str, Any] | None, filename: str) -> bool:
+def autogen_cargo(conf_file, yaml: Dict) -> Generator[Path]:
+    def render_stage(stage_conf: Mapping[str, Any] | None, filename: str) -> Generator[Path]:
```
It would be nice to comment what these return, now that it's less trivial.

kkysen replied:
Fixed in 3dea6a9.

fw-immunant commented on tests/__init__.py:
```diff
@@ -21,9 +22,11 @@ class Test(object):
+        self.scripts = {
+            f.name for f in Path(directory).iterdir() if f.suffix == ".sh" and
+            (".gen" not in f.suffixes or any(f.samefile(script) for script in generated_scripts))
+        }
```
Yes, I think so.

kkysen committed:
test.py: add docs to `autogen*` fns explaining what they return/yield now

kkysen committed:
test.py: make logic for finding scripts clearer

We search for and find non-generated scripts (`.sh`, but not `.gen.sh`),
whereas we record all generated scripts, which we then combine to get all scripts.

fw-immunant reviewed (approved):
LGTM, thanks.

kkysen commented on tests/__init__.py:
```diff
@@ -21,12 +22,15 @@ class Test(object):
+    def __init__(self, directory: Path, generated_scripts: set[Path]):
+        # We only want scripts that have been generated by us now,
+        # but non `*.gen*` scripts we can search for.
+        non_gen_script_paths = { f for f in directory.iterdir() if f.suffix == ".sh" and ".gen" not in f.suffixes }
```
Yeah, it can definitely make more sense, but I'm not sure that's worth juggling around right now.

ahomescu replied:
Sounds good, leave it for later.

ahomescu commented:
@kkysen You should move this to the main c2rust repo.

kkysen commented:
* Moved to https://github.com/immunant/c2rust/pull/1488.
